src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -467,5 +501,19 @@ public override T GetResult<T>()
                 return result.ToObject<T>(this.jsonSerializer);
             }
         }
+
+        private class JsonArrayPool<T> : IArrayPool<T>
```

Comment:
It's not clear why you need this from just reading the code; but looks like IArrayPool is actually a NewtonSoft thing and you created this as a shim between System.Buffers.ArrayPool and NewtonSoft.Json.IArrayPool. If so, you may want to add a comment here to make that clear.

Reply:
Apparently so, since both you and @dgriffen had the same concern.

Reply:
I've added an xml doc comment on the class.